docs/modules/ROOT/pages/cluster-side-modules.adoc

```diff
@@ -0,0 +1,153 @@
+= Working with Cluster-Side Modules in Hazelcast Cloud
+:description: Cluster-side modules are Java classes that a Hazelcast Cloud cluster can execute or store. You can write cluster-side modules to execute custom code or store custom objects. This guide describes the workflows and best practices for implementing cluster-side modules for Hazelcast Cloud.
+
+{description}
+
+The basic workflow for any cluster-side module is the following:
+
+. *Build* the Java class and implement any required interfaces.
+. *Test* your implementation in developer mode for fast deployment and easy access to cluster logs.
+. *Deploy* your Java classes to Hazelcast Cloud.
+
+Then you can use any client to interact with the cluster-side modules such as querying objects or triggering processes.
+
+image:cluster-side-modules.svg[Workflow for building cluster-side modules]
+
+== What are Cluster-Side Modules
+
+Cluster-side modules include any code that must be deployed as Java code to the cluster, including the following:
+
+.Cluster-side modules
+[cols="1a,2a"]
+|===
+|Module|Description
+
+|link:https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/map/MapStore.html[MapStore]
+|Ingest data from an external data source into a Hazelcast map, or add data from a Hazelcast map to an external data source. xref:mapstore-mongodb.adoc[See a tutorial].
+
+|link:https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/map/EntryProcessor.html[Entry processor]
+|Atomically execute code on a map entry.
+
+You cannot use a custom name for entry processors in Hazelcast Cloud. You must use the one called `default`.
+
+|link:https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/core/IExecutorService.html[Executor]
+|Asynchronously execute tasks, such as database queries, complex calculations, and image rendering.
+
+You cannot use a custom name for executors in Hazelcast Cloud. You must use the one called `default`.
+
+|Custom object class (domain objects)
+|Allow Hazelcast Cloud to store or process domain objects that you use in your applications.
+
+|link:https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/jet/pipeline/Pipeline.html[Pipeline]
+|Process streaming or batch data in Hazelcast Cloud, using one or more data sources and sinks.
+
+|===
+
+[[serializers]]
+== Supported Serializers
+
+Cluster-side modules need to be sent to other members in the cluster or one of your client applications. As a result, these modules need to be serialized.
+
+Hazelcast Cloud supports the following serializers. Follow the links to learn more about these serializers in the Hazelcast Platform documentation.
```

Comment:
Maybe it'd be better to change the order of the serializers listed. We want to focus on compact serialization as the default for the cloud. So maybe a different order (Compact, JsonValues and the others) would direct customers towards the better choices.